test/e2e/networking.go

```diff
@@ -101,155 +102,212 @@ var _ = Describe("Networking", func() {
+			var svcname = fmt.Sprintf("nettest-%v", port)
+
+			By(fmt.Sprintf("Creating a service named %q in namespace %q", svcname, f.Namespace.Name))
+			svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
+				ObjectMeta: api.ObjectMeta{
+					Name: svcname,
+					Labels: map[string]string{
+						"name": svcname,
+					},
+				},
+				Spec: api.ServiceSpec{
+					Ports: []api.ServicePort{{
+						Protocol:   "TCP",
+						Port:       port,
+						TargetPort: util.NewIntOrStringFromInt(port),
+					}},
+					Selector: map[string]string{
+						"name": svcname,
+					},
+				},
+			})
 			if err != nil {
-				Logf("Attempt %v: service/pod still starting. (error: '%v')", i, err)
-				continue
-			}
-			// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
-			switch {
-			case string(body) == "pass":
-				Logf("Passed on attempt %v. Cleaning up.", i)
-				passed = true
-			case string(body) == "running":
-				Logf("Attempt %v: test still running", i)
-			case string(body) == "fail":
-				if body, err = getDetails(); err != nil {
-					Failf("Failed on attempt %v. Cleaning up. Error reading details: %v", i, err)
-				} else {
-					Failf("Failed on attempt %v. Cleaning up. Details:\n%s", i, string(body))
-				}
-			case strings.Contains(string(body), "no endpoints available"):
-				Logf("Attempt %v: waiting on service/endpoints", i)
-			default:
-				Logf("Unexpected response:\n%s", body)
+				Failf("unable to create test service named [%s] %v on port %v", svc.Name, err, port)
+			} else {
+				Logf("************************** Created service successfully [%s]", svc.Name)
 			}
-		}
 
-		if !passed {
-			if body, err = getDetails(); err != nil {
-				Failf("Timed out. Cleaning up. Error reading details: %v", err)
-			} else {
-				Failf("Timed out. Cleaning up. Details:\n%s", string(body))
+			Logf("launching pod per node.....")
+			podNames := launchNetTestPodPerNode(port, nettestVersion, f, nodes, svcname)
+
+			Logf("************************** Launched test pods for %v", port)
+			By("Waiting for all of the webserver pods to transition to Running state")
+			for _, podName := range podNames {
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred())
+				By(fmt.Sprintf("Waiting for connectivity to be verified [ port =  %v ] ", port))
+				//once response OK, evaluate response body for pass/fail.
+				pollPeerStatus(sem, f, svc)
+
 			}
-		}
-		Expect(string(body)).To(Equal("pass"))
-	})
 
-})
+			Logf("************************** Finished test pods for %v", port)
+		}(nodes, ports[ii])
+	}
+	//now wait for the all X nettests to complete...
+	//By("Waiting for all connectivities to be verified")
+	for ii := range ports {
+		Logf("Now waiting on port %v", ports[ii])
+		<-sem
+		Logf("finished ! port %v", ports[ii])
+	}
+	Logf("Completed test on %v port/services", len(ports))
+}
+
+//Makeports makes a bunch of ports from 8080->8080+n
```

Comment:
given the use of static ports, we probably need to add this to the list of not-parallel-safe tests in `hack/jenkins/e2e.sh`.